soundfile.py

```diff
@@ -1332,10 +1332,11 @@ def _check_dtype(self, dtype):
 
     def _array_io(self, action, array, frames):
         """Check array and call low-level IO function."""
-        if (array.ndim not in (1, 2) or
-                array.ndim == 1 and self.channels != 1 or
-                array.ndim == 2 and array.shape[1] != self.channels):
-            raise ValueError("Invalid shape: {0!r}".format(array.shape))
+        if self.channels not in (1,2):
```

Comment:
This seems wrong. Previously, we were checking if the input array had one or two dimensions. Now we're checking whether we have one or two channels. Those are different things, right?

We support arbitrary numbers of channels (depending on the file type), but only one- or two-dimensional data.

Reply:
I misunderstood what the old code was doing but I see the problem now. Will fix